src/cloud-api-adaptor/pkg/userdata/provision.go

```diff
@@ -2,45 +2,52 @@ package userdata
 
 import (
 	"context"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/base64"
+	"encoding/hex"
 	"fmt"
 	"log"
 	"os"
 	"path/filepath"
+	"strings"
 	"time"
 
 	"github.com/avast/retry-go/v4"
 	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-providers/aws"
 	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-providers/azure"
 	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-providers/docker"
+	toml "github.com/pelletier/go-toml/v2"
 	"gopkg.in/yaml.v2"
 )
 
-var logger = log.New(log.Writer(), "[userdata/provision] ", log.LstdFlags|log.Lmsgprefix)
+const (
+	ConfigParent = "/run/peerpod"
+	DigestPath   = "/run/peerpod/initdata.digest"
+	InitdataPath = "/run/peerpod/initdata"
+)
 
-type paths struct {
-	aaConfig     string
-	agentConfig  string
-	authJson     string
-	cdhConfig    string
-	daemonConfig string
-}
+var logger = log.New(log.Writer(), "[userdata/provision] ", log.LstdFlags|log.Lmsgprefix)
+var WriteFilesList = []string{"agent-config.toml", "daemon.json", "auth.json", "initdata"}
```

Comment:
Maybe we can import those paths from the adapator/cloud (or the other way around, export the paths consts from here and use them in adaptor/cloud). I would also suggest using the full path (see comment below)

Reply:
Yes, we can let userdata depends on cloud.